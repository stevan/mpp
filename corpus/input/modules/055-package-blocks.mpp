# Package block with simple name
package Foo {
    my $x = 10;

    sub bar {
        return $x;
    }
}

# Package block with qualified name
package My::Module {
    my $value = 42;

    sub get_value {
        return $value;
    }
}

# Code after package block
my $y = 20;
