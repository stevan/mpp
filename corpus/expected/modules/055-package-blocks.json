[
  {
    "type": "Package",
    "name": "Foo",
    "body": [
      {
        "type": "Declaration",
        "declarator": "my",
        "variable": {
          "type": "Variable",
          "name": "$x"
        },
        "initializer": {
          "type": "Number",
          "value": "10"
        }
      },
      {
        "type": "Sub",
        "name": "bar",
        "parameters": [],
        "body": [
          {
            "type": "Return",
            "value": {
              "type": "Variable",
              "name": "$x"
            }
          }
        ]
      }
    ]
  },
  {
    "type": "Package",
    "name": "My::Module",
    "body": [
      {
        "type": "Declaration",
        "declarator": "my",
        "variable": {
          "type": "Variable",
          "name": "$value"
        },
        "initializer": {
          "type": "Number",
          "value": "42"
        }
      },
      {
        "type": "Sub",
        "name": "get_value",
        "parameters": [],
        "body": [
          {
            "type": "Return",
            "value": {
              "type": "Variable",
              "name": "$value"
            }
          }
        ]
      }
    ]
  },
  {
    "type": "Declaration",
    "declarator": "my",
    "variable": {
      "type": "Variable",
      "name": "$y"
    },
    "initializer": {
      "type": "Number",
      "value": "20"
    }
  }
]